src/Markdown/Block.elm

```diff
@@ -41,48 +58,177 @@ In the simplest case, you can pass this directly to a renderer:
 
     main : Html msg
     main =
-        case astResult of
-            Ok ast ->
-                Markdown.Parser.renderDefaultHtml ast
+        case
+            astResult
+                |> Result.mapError deadEndsToString
+                |> Result.andThen (\ast -> Markdown.Renderer.render Markdown.Renderer.defaultHtmlRenderer ast)
+        of
+            Ok rendered ->
+                div [] rendered
 
             Err errors ->
-                Html.text "Encountered parsing errors."
+                text errors
 
 -}
 type Block
-    = Heading Int (List Inline)
-    | Body (List Inline)
-    | Html String (List Attribute) (List Block)
-    | UnorderedListBlock
-        (List
-            { task : Maybe Bool
-            , body : List Inline
-            }
-        )
-    | OrderedListBlock Int (List (List Inline))
-    | CodeBlock Markdown.CodeBlock.CodeBlock
-    | ThematicBreak
+    = -- Container Blocks
+      HtmlBlock (Html Block)
+    | UnorderedList (List (ListItem Inline))
+    | OrderedList Int (List (List Inline))
```

Comment:
It might help the readability/understanding of this type to add a type (even just an alias) around the deeper nested list. It's a bit confusing at first glance. the `ListItem` for the `UnorderedList` helps a lot.